### PR TITLE
CSD 224 Install event - need to set required fields before the tracker is initialized

### PR DESCRIPTION
**Pull request by LamarDataSherpas**
https://thedatasherpas.atlassian.net/browse/CSD-224

### CODE REVIEW

File: SiftSdkTests/PlanTesting.swift [hunk 1]
```diff
@@ -24,9 +24,17 @@ class PlanTesting: XCTestCase {
...
+      requiredProperties.accountId = "AccountId 2:31:25"
+      requiredProperties.appVer = "someVersion"
+      requiredProperties.deviceId = "myDeviceId"
+      requiredProperties.eventPayload = [ "key": "value" ]
+      requiredProperties.partnerId = "The PartnerId"
+      requiredProperties.platform = "some plat"
+      try SiftSdk.shared.set(host: "44.234.38.156", protocol: .http, requiredProperties: requiredProperties)
     } catch {
       print(error)
+      print("#function:\(#line)")
```
Comment (QasimAbbas) [inline SiftSdkTests/PlanTesting.swift:37]: Should there be an XCTFail() used here?

File: SiftSdkTests/PlanTesting.swift [hunk 2]
```diff
@@ -88,7 +96,12 @@ class PlanTesting: XCTestCase {
 
     eventPayload["dict"] = dict
 
-    SiftSdk.shared.tagEvent(name: "EventName", customPayload: customPayload, eventPayload: eventPayload)
+    do {
+      try SiftSdk.shared.tagEvent(name: "EventName", customPayload: customPayload, eventPayload: eventPayload)
+    } catch {
+      print("Error: \(error)")
```
Comment (QasimAbbas) [inline SiftSdkTests/PlanTesting.swift:102]: Should there be an XCTFail() used here?

**Review by QasimAbbas — CHANGES_REQUESTED** [review]

File: SiftSdkTests/PlanTesting.swift (hunk 1 above)
Comment (LamarDataSherpas) [reply SiftSdkTests/PlanTesting.swift:37]: Technically yes, this whole test is a quick hack until we have support for http status code which you implemented in another PR. I will fix this when I start using the http status code code you implemented.

Comment (LamarDataSherpas) [reply SiftSdkTests/PlanTesting.swift]: done

File: SiftSdkTests/PlanTesting.swift (hunk 2 above)
Comment (LamarDataSherpas) [reply SiftSdkTests/PlanTesting.swift]: done

**Review by QasimAbbas — APPROVED** [review]

**Pull request merged**